docs/kubeconfig-file.md

```diff
@@ -46,14 +46,46 @@ users:
     token: blue-token
 - name: green-user
   user:
+    username: admin
+    password: secret
     client-certificate: path/to/my/client/cert
     client-key: path/to/my/client/key
 ```
 
+## .kubernetes_auth files
+
+The kubernetes_auth file is a legacy config file that can contain a mix of server and client auth info. It is supported in .kubeconfig via `auth-path` for a user:
+```
+users:
+- name: black-user
+  user:
+    auth-path: path/to/my/existing/.kubernetes_auth_file
+```
+
+The entries in a file loaded via auth-path will be applied to both the user and cluster of the current context.
+
+**WARNING**: merging auth from a mixture of kubernetes_auth file entries and .kubeconfig user entries is hard to debug and should be avoided. kubernetes_auth file support exists mostly for tests and is being deprecated.
+
+### Example .kubernetes_auth file
+```
+{
+  "User": "admin",
+  "Password": "secret",
+  "CertFile": "/path/to/my/client/cert",
+  "KeyFile": "/path/to/my/client/key",
+  "CAFile": "/path/to/my/server/cafile",
+  "BearerToken": "secrettoken",
+  "Insecure": false
+}
+```
+All entries are optional. `User`, `Password`, `CertFile`, `KeyFile`, and `BearerToken` are applied to the kubectl user. `CAFile` and `Insecure` apply to the cluster. The same rules regarding one authentication technique apply as for .kubeconfig entries (see below).
```

Comment:
validateAuthInfo doesn't detect conflicting `BearerToken` and `User` authentication loaded from `--auth-path`: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/client/clientcmd/validation.go#L139